### DISCUSSION TITLE
Plugin development question

### DISCUSSION THREAD

**Discussion by MalfuncEddie** [netbox-community/netbox#9993] [Q&A] [answered]
Hi,

I'm trying to develop a plugin by following https://github.com/netbox-community/netbox-plugin-tutorial. At the end of https://github.com/netbox-community/netbox-plugin-tutorial/blob/main/tutorial/step05-views.md

You should be able to see the UI.
I get the following error

```
File "/netbox/releases/netbox-3.2.2/venv/lib/python3.10/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/netbox/releases/netbox-3.2.2/venv/lib/python3.10/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/netbox/releases/netbox-3.2.2/venv/lib/python3.10/site-packages/django/views/generic/base.py", line 84, in view
    return self.dispatch(request, *args, **kwargs)
  File "/netbox/netbox/netbox/utilities/views.py", line 87, in dispatch
    if not self.has_permission():
  File "/netbox/netbox/netbox/utilities/views.py", line 73, in has_permission
    self.queryset = self.queryset.restrict(user, action)

Exception Type: AttributeError at /plugins/datadump/datadumps/
Exception Value: 'QuerySet' object has no attribute 'restrict'

```
Anyone know why this is. I am starting to think this is an old tutorial? Because it seems that I do not have permissions to do this (I am superuser)

**Comment by jeremystretch** [comment, marked as answer]
> Anyone know why this is. I am starting to think this is an old tutorial?

Nope, it's quite current.

> Because it seems that I do not have permissions to do this (I am superuser)

The exception has nothing to do with permissions. The queryset you've defined for your view references a model that does not employ the RestrictedQuerySet manager. Compare your work to the branch for step 5 in the tutorial repo to see what's missing.

**Reply by MalfuncEddie** [reply]
I think I found where I went wrong. I did not use NetBoxModel but only certain features.  
Since only NetBoxModel  and ChangeLoggedModel do "objects = RestrictedQuerySet.as_manager()" and I did not include ChangeLoggedModel  in the features it does not work.

side note:
The plugin I was trying to do has as function to "Dump data" linked to a vm without creating a changelog entry. I got a backend script that creates a large json per vm  that changes every day and it would be nice if I could add this into netbox without polluting the changelog. I'll try and stumble around a bit more since it is fun learning. 


Thank you for your comment and your continued work on Netbox. This project has made me pick up Django(very noob stage) and rekindled web dev instead of CLI scripting.
